packages/main/src/FileUploader.ts

```diff
@@ -317,6 +360,33 @@ class FileUploader extends UI5Element implements IFormInputElement {
 		}, "");
 	}
 
+	/**
+	 * Checks whether all files are below `maxFileSize` (if set),
+	 * and fires a `fileSizeExceeded` event if any file exceeds it.
+	 *
+	 * @returns true if all files are below `maxFileSize`
+	 * @private
+	 */
+	_validateFileSize(files: FileList | null): boolean {
+		if (!this.maxFileSize || !files) {
+			return true;
+		}
+
+		const convertBytesToMegabytes = (bytes: number) => (bytes / 1024) / 1024;
+
+		for (let i = 0; i < files.length; i++) {
```

Comment:
It's a matter of preference (no need to change it) but generally it's best to use `forEach` because it's less and more readable code, you won't have the `[]` and you won't have an index `i`. The `for` operator should be used only if you must call `return` inside to return from the whole function (as opposed to `forEach` where return continues to the next iteration of the loop)

Reply:
Resolving this because, as it was discussed below, there is a return to prevent multiple events from being fired... But this may be changed shortly.